### PR TITLE
Yet more random tests

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/floatfuncs.jl [hunk 1]
```diff
@@ -68,3 +68,5 @@ for elty in (Float32,Float64)
         @test round(elty2,A) == fill(round(elty2,x),(10,10,10))
     end
 end
+
+@test float(Char) == Any
```
Comment (tkelman) [inline test/floatfuncs.jl]: what needs this?

Comment (kshyatt) [inline test/floatfuncs.jl]: Well, the function is defined [here](https://codecov.io/github/kshyatt/julia/base/float.jl?ref=a92c7ff518408182d2bcdf3cbd0dbe3f2c1c1f40#l-125).

Comment (IainNZ) [inline test/floatfuncs.jl]: Jeez thats weird

Comment (tkelman) [inline test/floatfuncs.jl]: what would break if we got rid of it?

Comment (StefanKarpinski) [inline test/floatfuncs.jl]: That may be a vestige from when `Char` was a type of number. It isn't anymore, so we should delete this definition.

**Comment by kshyatt** [comment]
Hopefully I did the right thing here...

**Comment by tkelman** [comment]
you'll also need to get rid of your test for it, I would assume

**Comment by kshyatt** [comment]
@tkelman better?

**Comment by ivarne** [comment]
The float method was introduced in #14413 (quite recently), and seems to [still be in use](https://github.com/JuliaLang/julia/blob/9ee9aefe814e6a11992983285db02a154a3ed17c/base/statistics.jl#L581) (although untested). 

cc @simonbyrne 

PS: There seems to be some confusion in the comments here because @kshyatt tested ``float(Char)``, but the method definition applies to all non `Number` types (``float{T}(::Type{T})``). This has nothing specifically to do with `Char`

**Comment by tkelman** [comment]
whoops, should have checked blame. and there should've been a test that needed it? what does the Any result allow that to do though?

**Pull request merged**